snakemake_storage_plugin_gcs/__init__.py

```diff
@@ -525,7 +525,7 @@ def _download_directory(self):
         self.local_path().mkdir(exist_ok=True)
 
         for blob in self.directory_entries():
-            local_name = f"{blob.bucket.name}/{blob.name}"
+            local_name = self.provider.local_prefix / self.bucket.name / blob.name
 
```

Comment:
Using a hard coded `/` also assumes a linux OS, which is generally OK, but it's more suitable to use `os.sep`

Reply:
@manbat I might be mistaken but since `self.provider.local_prefix` in `self.provider.local_prefix / self.bucket.name / blob.name` is of `pathlib.Path`, using `/` to construct paths should work independent of which os is executed on. 
I believe `pathlib` will handle the posix path generation from there.